src/components/WfsLayer.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react'
+import L, { GeoJSONOptions } from 'leaflet'
+import { FeatureCollection } from 'geojson'
+
+import { useMapInstance, GeoJSON } from '@datapunt/react-maps'
+import fetchWithAbort from '../utils/fetchWithAbort'
+import { ZoomLevel } from '../types'
+import { MIN_ZOOM_LEVEL, MAX_ZOOM_LEVEL } from '../constants'
+
+const NO_DATA: FeatureCollection = {
+  type: 'FeatureCollection',
+  features: [],
+}
+
+interface Props {
+  url: string
+  options?: GeoJSONOptions
+  zoomLevel: ZoomLevel
+}
+
+const getBBox = (mapInstance: any) => {
+  return `&${L.Util.getParamString({
+    bbox: mapInstance.getBounds().toBBoxString(),
+  }).substring(1)}`
+}
+
+const isVisible = (mapInstance: any, zoomLevel: ZoomLevel) => {
+  const { min, max } = zoomLevel
+  const zoom = mapInstance.getZoom()
+  return zoom <= (min || MIN_ZOOM_LEVEL) && zoom >= (max || MAX_ZOOM_LEVEL)
+}
+
+/**
+ * WfsLayer component should be only used when the api provides wfs capabilities for spatial queries
+ * The data is requested on each map action (pan, zoom)
+ *
+ */
+const WfsLayer: React.FC<Props> = ({ url, options, zoomLevel }) => {
+  const mapInstance = useMapInstance()
+  const [layerInstance, setLayerInstance] = useState()
```

Comment:
can pass GeoJSON type?

Reply:
Done, it conflicts with the GeoJSON react type from react-maps though.